### ISSUE TITLE
after fresh kubeadm init using flannel coredns is stuck in  CrashLoopBackOff status

### ISSUE THREAD

**Issue by cramakanth** [flannel-io/flannel#1281]
I installed kubeadm following instructions given here: https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/

  on 1 master and 2 worker nodes in a cluster and on the master did a kubeadm init with --pod-network-cidr=10.244.0.0/16 and --apiserver-advertise-address of my master node.

It went well and after finishing init, I then run the command and I get the corresponding output:

kubectl get pods --all-namespaces
NAMESPACE     NAME                                 READY   STATUS             RESTARTS   AGE
kube-system   coredns-66bff467f8-5fqcc             0/1     CrashLoopBackOff   16         40m
kube-system   coredns-66bff467f8-bxrlp             0/1     CrashLoopBackOff   16         40m
kube-system   etcd-kubemaster                      1/1     Running            1          41m
kube-system   kube-apiserver-kubemaster            1/1     Running            1          41m
kube-system   kube-controller-manager-kubemaster   1/1     Running            1          41m
kube-system   kube-flannel-ds-amd64-7w5jz          1/1     Running            1          37m
kube-system   kube-proxy-hhlws                     1/1     Running            1          40m
kube-system   kube-scheduler-kubemaster            1/1     Running            1          41m

A you can see coredns never comes out of CrashLoopBackOff status. Cannot proceed to join the worked nodes.

* Flannel version: v0.11.0
* Backend used (e.g. vxlan or udp): Don;t know. Looks to be udp
* Etcd version:
* Kubernetes version (if used):
* Operating System and version: Description:    Ubuntu 16.04.6 LTS
Release:        16.04
Codename:       xenial
* Link to your project (optional):

**Comment by changsen-dev** [comment]
what's codedns pod's log?

**Comment by Gacko** [comment]
10.96.0.1 should get translated to a host network address and therefore not be routed through Flannel interface.

**Comment by Gacko** [comment]
```--apiserver-advertise-address``` probably isn't what you except it to be. See the documentation for ```--control-plane-endpoint``` if you're trying to set the cluster address for our API server endpoint.

**Comment by chief93** [comment]
@Gacko first of all - thank for you suggestions of `--control-plane-endpoint` flag, because I NEVER get to this flag in any docs or other suggestions about solving same issues.
Next - both within and without `--apiserver-advertise-address` flag Flannel isn't able to get up and running. And one more evidence about the fact the Flannel is wrong, that other CNI plugin - `kube-router` - is up and running right within same setup procedure - it can connect to the API through it's public address and set the networking up and running without any additional config patches or something like this.
I am unable to provide logs about the `kube-proxy` because of [this issue](https://github.com/kubernetes/kubernetes/issues/90083) which drove me to delete the `kube-proxy` from the cluster. And seems like cluster is up and running fine as expected, right even without it. Also, unfortunately, I'm unable to provide the logs of the Flannel itself, because as I said, I found a working solution. I deleted Flannel from the cluster, because it's failure blocked the `core-dns` startup.

And I'm sorry, but one another big disappointment in Flannel, that when it is up and running, it spaming the ARP table of upstream router (Mikrotik in my case), which leads to blocking of ALL incoming traffic to the hypervisor machine. I think that this issue can be somehow dependent of current issue subject.

Anyway, thanks for Flannel, besides I was able to start it, within config patches, at my firsts Kubernetes setup attempts.
